packages/lesswrong/components/tagging/subforums/RightSidebarColumn.tsx

```diff
@@ -98,30 +117,89 @@ export const styles = (theme: ThemeType): JssStyles => ({
   gap2: { gridArea: 'gap2' }
 });
 
-export const RightSidebarColumn = ({header, sidebarComponents = [], children, classes}: {
-  header: React.ReactNode,
+export const RightSidebarColumn = ({titleComponent, bannerImageId, headerComponent, sidebarComponents = [], children, classes}: {
+  titleComponent: React.ReactNode,
+  bannerImageId: string,
+  headerComponent: React.ReactNode,
   children: React.ReactNode,
   classes: ClassesType,
   sidebarComponents?: React.ReactNode[],
 }) => {
   const nonEmptySidebarComponents = sidebarComponents.filter(x => x) // filter out nulls to avoid extra spacing
+  const { CloudinaryImage2 } = Components
+  
+  /*
+   * The logic for rendering the banner image has turned out more complicated than I would
+   * have liked, but unfortunately it is neccesary to achieve the following things:
+   * 1. Cropping the image nicely
+   * 2. Not having any noticeable layout shift (on mobile)
+   *
+   * Explanation:
+   * 1. We want the title div to be the exact same height as the background image, the best
+   *    way to do this would be to set the image in CSS as the `background-image` of the title div.
+   *    Unfortunately there is a lot of magic in CloudinaryImage2 which makes the cropping look
+   *    nice, I couldn't reproduce this with the `background-image` method so I have ended up
+   *    setting the title div as `position: relative` and the background image (and translucent overlay)
+   *    as `position: absolute` to make them display on top of each other
+   * 2. The downside of doing the `position: relative`/`position: absolute` trick is that there is
+   *    no way for the two divs to know each others height from pure CSS. Additionally, the cropping
+   *    in CloudinaryImage2 relies on knowing the height in code, so it's ~necessary to set the height
+   *    in code.
+   *    This can't be done until after the page has rendered once, which would cause a layout
+   *    shift. To fix this I have made it so the banner image starts off slightly oversized but is hidden
+   *    behind the next element by using z-indices. Once the page has rendered once, the height gets
+   *    updated to exactly the right value which fixes the cropping (this is fairly discreet)
+   */
```

Comment:
I found this kind of hard to follow, but it's also the end of the day on Friday so I'm not the most focused. So you're saying you wanted to use `background-image` but are not currently using that?

All the hard-coding in this file isn't my favorite, but it does seem to work well so I'm happy to keep it for now. It doesn't feel necessary for me to totally dive into this. If you'd like me to do so in the future, I can make some time for that. :) For now I'll just say, another way to make the text stick to the bottom of the image is to put the text in a flex container in the column direction (I didn't read closely enough to know if this would be helpful for you, but thought I'd mention it just in case).

Reply:
Thanks, I'll remember the flex trick for next time. I did try a lot of different approached here and ended up with this fairly complex one, which I'm not that happy with. But it does do the job at least